runtime/src/accounts.rs

```diff
@@ -623,18 +678,17 @@ impl Accounts {
             {
                 if message.is_debitable(i) {
                     accounts.push((key, account));
-                }
-                if *credit > 0 {
-                    // Increment credit-only account balance Atomic
+                } else {
                     self.credit_only_account_locks
-                        .read()
+                        .write()
                         .unwrap()
-                        .as_ref()
+                        .as_mut()
                         .expect("Collect accounts should only be called before a commit, and credit only account locks should exist before a commit")
-                        .get(key)
-                        .unwrap()
-                        .credits
-                        .fetch_add(*credit, Ordering::Relaxed);
+                        .entry(*key)
+                        .and_modify(|lock| {
+                            lock.rent_debtor = true;
```

Comment:
this boolean is really "a commit occurred", right?

Reply:
this boolean 

> have you root-caused the issue that Stephen observed? can you explain it to me?

Since tx batches are executed in parallel, what happened that `take_credit_only` function which replaces actual hashmap with None, was executed and the new code, since it didn't have any safeguard failed, because hashmap was None

Reply:
right, but accounts generally owe rent (right?) this just says "yeah, we committed, so run rent on this guy at the end of the slot"

Reply:
So, CO account won't owe any rent in case tx fails, but, will still be part of `credit_only_lock` hashmap. So, when the boolean is true it means this account was used in successful txs and it ows rent.